### PR TITLE
Adopt cmake to Qt6 build

### PR DESCRIPTION
**Pull request by s-lisovenko**
At the moment we can successfully compile the program, but when launched, the program starts incorrectly with errors @DonLakeFlyer @patrickelectric Pplease could you please take a look at this MR and help fix it?

This is what i see after launch:
image

And this is an example of error output:

qrc:/qml/QGroundControl/ScreenTools/ScreenTools.qml:113:5: QML Connections: Detected function "onValueChanged" in Connections element. This is probably intended to be a signal handler but no signal of the target matches the name.
qrc:/qml/QGroundControl/ScreenTools/ScreenTools.qml:190: TypeError: Cannot read property 'appFontPointSize' of undefined
qrc:/qml/QGroundControl/FlightMap/FlightMap.qml:113:5: QML Connections: Detected function "onRawValueChanged" in Connections element. This is probably intended to be a signal handler but no signal of the target matches the name.
qrc:/qml/QGroundControl/FlightMap/FlightMap.qml:108:5: QML Connections: Detected function "onRawValueChanged" in Connections element. This is probably intended to be a signal handler but no signal of the target matches the name.
qrc:/qml/QGroundControl/FlightMap/FlightMap.qml:113:5: QML Connections: Detected function "onRawValueChanged" in Connections element. This is probably intended to be a signal handler but no signal of the target matches the name.
qrc:/qml/QGroundControl/FlightMap/FlightMap.qml:108:5: QML Connections: Detected function "onRawValueChanged" in Connections element. This is probably intended to be a signal handler but no signal of the target matches the name.
qrc:/qml/QGroundControl/FlightMap/FlightMap.qml:89: TypeError: Cannot read property 'mapProvider' of undefined
qrc:/qml/QGroundControl/FlightMap/FlightMap.qml:89: TypeError: Cannot read property 'mapProvider' of undefined
qrc:/FirstRunPromptDialogs/UnitsFirstRunPrompt.qml:32: TypeError: Cannot read property 'length' of undefined
setCurrentPlanViewSeqNum
setCurrentPlanViewSeqNum
qrc:/qml/QGroundControl/FlightMap/MapScale.qml:124: TypeError: Cannot read property 'horizontalDistanceUnits' of undefined
qrc:/qml/MissionSettingsEditor.qml:34: TypeError: Cannot read property 'missionFileExtension' of undefined
qrc:/qml/MissionSettingsEditor.qml:33: TypeError: Cannot read property 'missionSavePath' of undefined
qrc:/qml/MissionSettingsEditor.qml:107: TypeError: Cannot read property 'defaultMissionItemAltitude' of undefined
qrc:/qml/MissionSettingsEditor.qml:176: TypeError: Cannot read property 'offlineEditingFirmwareClass' of undefined
qrc:/qml/MissionSettingsEditor.qml:192: TypeError: Cannot read property 'offlineEditingVehicleClass' of undefined
qrc:/qml/MissionSettingsEditor.qml:218: TypeError: Cannot read property 'offlineEditingCruiseSpeed' of undefined
qrc:/qml/MissionSettingsEditor.qml:229: TypeError: Cannot read property 'offlineEditingHoverSpeed' of undefined
qrc:/qml/QGroundControl/Controls/FirstRunPrompt.qml:18: ReferenceError: Dialog is not defined
qrc:/FirstRunPromptDialogs/UnitsFirstRunPrompt.qml:87: TypeError: Cannot read property 'length' of undefined
qrc:/FirstRunPromptDialogs/UnitsFirstRunPrompt.qml:98: TypeError: Cannot read property 'horizontalDistanceUnits' of undefined
qrc:/FirstRunPromptDialogs/UnitsFirstRunPrompt.qml:103: TypeError: Cannot read property 'length' of undefined
qrc:/FirstRunPromptDialogs/UnitsFirstRunPrompt.qml:26: TypeError: Cannot read property 'horizontalDistanceUnits' of undefined
qrc:/qml/QGroundControl/Controls/LogReplayStatusBar.qml:32: TypeError: Cannot read property 'telemetrySavePath' of undefined

### CODE REVIEW

**Comment by s-lisovenko** [comment]
Video streaming is currently disabled by default in cmake, to enable it you need to add -DENABLE_VIDEOSTREAMING=ON to cmake.

File: src/FirmwarePlugin/APM/APMFirmwarePlugin.cc [hunk 1]
```diff
@@ -29,7 +29,7 @@
 #include "LinkManager.h"
 
 #include <QTcpSocket>
-#include <QRegExp>
+#include <QtCore5Compat/QRegExp>
```
[the remark this answers is not in the document]
Comment (DonLakeFlyer) [reply src/FirmwarePlugin/APM/APMFirmwarePlugin.cc:32]: Ok, can't hurt to be more specific I guess.

[the remark this answers is not in the document]
Comment (s-lisovenko) [reply src/FirmwarePlugin/APM/APMFirmwarePlugin.cc:32]: But maybe this is redundant, i don't know. We can drop this commit if you want

[the remark this answers is not in the document]
Comment (DonLakeFlyer) [reply src/FirmwarePlugin/APM/APMFirmwarePlugin.cc:32]: I think it's worthwhile for folks to understand this is a Qt5 thing. Eventually we should update the code to QRegularExpression.

**Comment by s-lisovenko** [comment]
Now we can compile QGC using cmake. There are currently some qml errors related to qml types associated with video controls.

**Comment by s-lisovenko** [comment]
Now old cmake build works with Qt6. 
To compile, you need to replace Qt5::Network in the CMakeLists.txt file with Qt6::Network in the src folder for the qmdnsengine library.

**Comment by DonLakeFlyer** [comment]
> To compile, you need to replace Qt5::Network in the CMakeLists.txt file with Qt6::Network in the src folder for the qmdnsengine library.

Could you submit a pull against qmdnsengine to fix that?

**Comment by DonLakeFlyer** [comment]
I plan to merge this into the Qt6 branch tonight and then clean up the Qt6 branch. Planning to get it ready to merge in to master tomorrow.

**Comment by DonLakeFlyer** [comment]
And thanks for all the work on this.

**Comment by s-lisovenko** [comment]
> I plan to merge this into the Qt6 branch tonight and then clean up the Qt6 branch. Planning to get it ready to merge in to master tomorrow.

Please wait one hour, i found strange behavior

**Comment by s-lisovenko** [comment]
> > To compile, you need to replace Qt5::Network in the CMakeLists.txt file with Qt6::Network in the src folder for the qmdnsengine library.
> 
> Could you submit a pull against qmdnsengine to fix that?

I will try

**Comment by DonLakeFlyer** [comment]
> Please wait one hour, i found strange behavior

Ok, let me know and then I'll merge.

**Comment by s-lisovenko** [comment]
@DonLakeFlyer I think I'll finish work for today, so you can merge it. I found strange behavior related UI scalling. For qmake i have 111% but for cmake scalling is 91%, but looks the same as for qmake.

**Comment by DonLakeFlyer** [comment]
Great thanks.

**Comment by s-lisovenko** [comment]
> > To compile, you need to replace Qt5::Network in the CMakeLists.txt file with Qt6::Network in the src folder for the qmdnsengine library.
> 
> Could you submit a pull against qmdnsengine to fix that?

I think we need just change head of this sub module, because original version supports qt6 to. I noticed that we use fork by @patrickelectric for this library

**Comment by DonLakeFlyer** [comment]
At some point we'll look at moving back to regular repo. Not sure why we're using a fork. But at this point safest/quickest thing is to update the forked version. 

If you have time you can try moving back to regular repo version, but if so make sure you test it all out for problems.

**Pull request merged**